development/cpp/internal_rendering_architecture.rst

```diff
@@ -0,0 +1,694 @@
+- `scene/3d/lightmapper.cpp <https://github.com/godotengine/godot/blob/master/scene/3d/lightmapper.cpp>`__ - Abstract class
+- `modules/lightmapper_rd/lightmapper_rd.cpp <https://github.com/godotengine/godot/blob/master/modules/lightmapper_rd/lightmapper_rd.cpp>`__ - GPU-based lightmapper implementation
+
+**Lightmapper GLSL shaders:**
+
+- `modules/lightmapper_rd/lm_raster.glsl <https://github.com/godotengine/godot/blob/master/modules/lightmapper_rd/lm_raster.glsl>`__
+- `modules/lightmapper_rd/lm_compute.glsl <https://github.com/godotengine/godot/blob/master/modules/lightmapper_rd/lm_compute.glsl>`__
+- `modules/lightmapper_rd/lm_blendseams.glsl <https://github.com/godotengine/godot/blob/master/modules/lightmapper_rd/lm_blendseams.glsl>`__
+
+Depth of field
+^^^^^^^^^^^^^^
+
+.. note::
+
+  Only available in the Clustered Forward and Forward Mobile backends, not the
+  Compatibility backend.
+
+The Clustered Forward and Forward Mobile backends use different approaches to
+DOF rendering, with different visual results. This is done to best
+match the performance characteristics of the target hardware. In Clustered
+Forward, DOF is performed using a compute shader. In Forward Mobile, DOF is
+performed using a fragment shader (raster).
+
+Box, hexagon and circle bokeh shapes are available (from fastest to slowest).
+Depth of field can optionally be jittered every frame to improve its appearance
+when temporal antialiasing is enabled.
+
+**Depth of field C++ code:**
+
+- `servers/rendering/renderer_rd/effects/bokeh_dof.cpp <https://github.com/godotengine/godot/blob/master/servers/rendering/renderer_rd/effects/bokeh_dof.cpp>`__
+
+**Depth of field GLSL shader (compute - used for Clustered Forward):**
+
+- `servers/rendering/renderer_rd/shaders/effects/bokeh_dof.glsl <https://github.com/godotengine/godot/blob/master/servers/rendering/renderer_rd/shaders/effects/bokeh_dof.glsl>`__
+
+**Depth of field GLSL shader (raster - used for Forward Mobile):**
+
+- `servers/rendering/renderer_rd/shaders/effects/bokeh_dof_raster.glsl <https://github.com/godotengine/godot/blob/master/servers/rendering/renderer_rd/shaders/effects/bokeh_dof_raster.glsl>`__
+
+Screen-space effects (SSAO, SSIL, SSR, SSS)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note::
+
+  Only available in the Clustered Forward backend, not the Forward Mobile or
+  Compatibility backends.
+
+The Clustered Forward backend supports screen-space ambient occlusion,
+screen-space indirect lighting, screen-space reflections and subsurface scattering.
+
+SSAO uses an implementation derived from Intel's
+`ASSAO <https://www.intel.com/content/www/us/en/developer/articles/technical/adaptive-screen-space-ambient-occlusion.html>`__
+(converted to Vulkan). SSIL is derived from SSAO to provide a high-performance
+implementation of screen-space global illumination (albeit with only indirect
+lighting simulated, hence its name).
```

Comment:
```suggestion
(converted to Vulkan). SSIL is derived from SSAO to provide high-performance
indirect lighting.
```

SSIL is explicitly not a form of global illumination, I would not call it one either as it will create unfair expectations.